src/plugins.ts

```diff
@@ -61,19 +60,18 @@ export default class Plugins {
     const {pluginPrefix, scope} = this.config.pjson.oclif
     if (!scope) return name
     const match = name.match(`@${scope}/${pluginPrefix ?? 'plugin'}-(.+)`)
-    if (!match) return name
-    return match[1]
+    return match?.[1] ?? name
   }
 
-  async hasPlugin(name: string): Promise<Interfaces.PJSON.PluginTypes.Link | Interfaces.PJSON.User | boolean> {
+  async hasPlugin(name: string): Promise<Interfaces.PJSON.PluginTypes.Link | Interfaces.PJSON.User | false> {
     const list = await this.list()
     const friendly = list.find((p) => this.friendlyName(p.name) === this.friendlyName(name))
     const unfriendly = list.find((p) => this.unfriendlyName(p.name) === this.unfriendlyName(name))
-    const link = list.find((p) => p.type === 'link' && path.resolve(p.root) === path.resolve(name))
+    const link = list.find((p) => p.type === 'link' && resolve(p.root) === resolve(name))
     return (friendly ?? unfriendly ?? link ?? false) as
       | Interfaces.PJSON.PluginTypes.Link
       | Interfaces.PJSON.User
```

Comment:
should this be `Interfaces.PJSON.PluginTypes.User` instead of `Interfaces.PJSON.User` ?  That's what `this.list` returns?  

seems to build and test OK, then the ASsertions aren't necessary.